packages/expect/src/utils.ts

```diff
@@ -251,6 +258,12 @@ export const iterableEquality = (
     return false;
   }
 
+  const aEntries = getObjectEntries(a);
+  const bEntries = getObjectEntries(b);
+  if (!equals(aEntries, bEntries)) {
```

Comment:
Do we need to add `[iterableEquality]` as optional `customTesters` argument to `equals` call in case any of these property values have iterators? All the calls in `matchers.ts` include it. I am asking, not telling.

This illustrates a limitation that we recently noticed: it is not (yet) possible to for a custom tester to know about sibling testers (for example, `toStrictEqual` matcher needs more testers than `toEqual` does, and `iterableEquality` has no way to provide them in this recursive call).

Reply:
@pedrottimark is there an extra test case you're thinking of we should add?

Reply:
I'm so sorry for the very late follow up.
I'm not familiar with the code base but I see the point and the limitation.

> This illustrates a limitation that we recently noticed: it is not (yet) possible to for a custom tester to know about sibling testers 

IMO `customTesters` should be kept like this so each custom tester should test only one thing, so my changes in `iterableEquality` method will be reverted and will include non-enumerable members again in `keys` method in `jasmineUtils.ts`
I'm not sure of the consequences of this change but will try it out.